File: laba2/go/2-8.go
package main

import (
	"fmt"
	"sort"
)

func main1() {
	var n int
	fmt.Println("Введите количество встреч:")
	fmt.Scan(&n)

	type Meeting struct {
		ID    int
		Start int
		End   int
	}
	var meetings []Meeting
	fmt.Println("Введите время начала и окончания каждой встречи (формат: start end): ")
	for i := 0; i < n; i++ {
		var start, end int
		fmt.Scan(&start, &end)
		meetings = append(meetings, Meeting{ID: i + 1, Start: start, End: end})
	}
	sort.Slice(meetings, func(i, j int) bool {
		return meetings[i].End < meetings[j].End
	})

	var result []int
	var count = 0
	var last_end = -1

	for i := 0; i < n; i++ {
		if meetings[i].Start >= last_end {
			result = append(result, meetings[i].ID)
			last_end = meetings[i].End
			count++
		}
	}
	fmt.Println("Количество выбранных встреч: ", count)
	fmt.Println("Номера встреч: ")
	for i := 0; i < len(result); i++ {
		fmt.Print(result[i]+1, " ")
	}

}

File: laba2/c++/2.8.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(){
    int n;
    cout << "Введите количество встреч: ";
    cin >> n;
    cout << "Введите время начала и окончания каждой встречи (формат: start end):" << endl;
    vector<pair<int, pair<int, int>>> meetings;
    for (int i = 0; i < n; i++){
        int start, end;
        cin >> start >> end;
        meetings.push_back({end, {start, i}});
    }
    
    sort(meetings.begin(), meetings.end());
    
    vector<int> result;
    int count = 0;
    int last_end = -1;
    
    for (int i = 0; i < n; i++){
        if (meetings[i].second.first >= last_end){
            count++;
            last_end = meetings[i].first;
            result.push_back(meetings[i].second.second);
        }
    }
    
    cout << "Количество выбранных встреч: " << count << endl;
    cout << "Номера выбранных встреч: ";
    for (int i = 0; i < result.size(); i++){
        cout << result[i] + 1 << " ";
    }
    cout << endl;

    return 0;
}